jdaviz/configs/imviz/plugins/viewers.py

```diff
@@ -92,7 +93,9 @@ def on_mouse_or_key_event(self, data):
             # Extract first dataset from visible layers and use this for coordinates - the choice
             # of dataset shouldn't matter if the datasets are linked correctly
             image = active_layer.layer
-            self.label_mouseover.icon = self.jdaviz_app.state.layer_icons.get(active_layer.layer.label)  # noqa
+            icon = self.jdaviz_app.state.layer_icons.get(active_layer.layer.label)
+            if icon is not None:
+                self.label_mouseover.icon = icon
```

Comment:
The icon doesn't work well for arbitrary BaseCartesianData instances at the moment, this needs to be investigated in glue-core but for now this ensures a graceful fallback if an icon is not found (which we should do anyway).

Reply:
We should probably set to an empty string or have the icon itself smart enough to handle None in `layer_viewer_icon.vue` (rather than just leave it at its previous value).  I'm not sure why `BaseCartesianData` would have any problem with the layer icon logic though, `layer_icons` gets populated in https://github.com/spacetelescope/jdaviz/blob/f36f3b2cca7d1f91eafb7220c4c4835af48e0a17/jdaviz/app.py#L384-L386 which should be called by any `AddDataMessage`.  Is it possible the `AddDataMessage` isn't being sent from glue for `BaseCartesianData`?